deployment/ccip/1_6_0/sequences/deploy_chain_contracts.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/smartcontractkit/chainlink-ccip/deployment/deploy"
+	"github.com/smartcontractkit/chainlink-ccip/deployment/utils/sequences"
+	cldf_chain "github.com/smartcontractkit/chainlink-deployments-framework/chain"
+	"github.com/smartcontractkit/chainlink-deployments-framework/chain/ton"
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/xssnick/tonutils-go/address"
+
+	tonops "github.com/smartcontractkit/chainlink-ton/deployment/ccip"
+	"github.com/smartcontractkit/chainlink-ton/deployment/ccip/helpers"
+	"github.com/smartcontractkit/chainlink-ton/deployment/ccip/operation"
+
+	ccipConfig "github.com/smartcontractkit/chainlink-ton/deployment/ccip/config"
+	seq "github.com/smartcontractkit/chainlink-ton/deployment/ccip/sequence"
+	"github.com/smartcontractkit/chainlink-ton/deployment/state"
+	"github.com/smartcontractkit/chainlink-ton/pkg/ton/tvm"
+)
+
+// defaultCCIPContractCoin is the default amount of TON coins to allocate for each CCIP contract deployment.
+// This value is chosen to cover contract initialization and storage costs on TON blockchain.
+const defaultCCIPContractCoin = "0.05"
+
+// defaultReserveAmount is the default reserve amount allocated to the OnRamp contract.
+// This reserve ensures the contract has sufficient balance for operational transactions.
+const defaultReserveAmount = "0.5"
+
+func (a *TonAdapter) DeployChainContracts() *operations.Sequence[deploy.ContractDeploymentConfigPerChainWithAddress, sequences.OnChainOutput, cldf_chain.BlockChains] {
+	return DeployChainContracts
+}
+
+var DeployChainContracts = operations.NewSequence(
+	"deploy-chain-contracts",
+	semver.MustParse("1.6.0"),
+	"Deploys all required contracts for CCIP 1.6.0 to a TON chain",
+	func(b operations.Bundle, chains cldf_chain.BlockChains, input deploy.ContractDeploymentConfigPerChainWithAddress) (output sequences.OnChainOutput, err error) {
+		tonChain := chains.TonChains()[input.ChainSelector]
+
+		// deps used for op
+		deps, err := extractTonDepsFromContractDeploymentInput(tonChain, input.ExistingAddresses)
+		if err != nil {
+			return sequences.OnChainOutput{}, err
+		}
+		seqInput, err := intoDeployCCIPSeqInput(input, deps.TonChain.WalletAddress)
+		if err != nil {
+			return sequences.OnChainOutput{}, err
+		}
+		ccipSeqReport, err := operations.ExecuteSequence(b, seq.DeployCCIPSequence, deps, seqInput)
+		if err != nil {
+			return sequences.OnChainOutput{}, fmt.Errorf("failed to deploy CCIP for TON chain %d: %w", input.ChainSelector, err)
+		}
+
+		deps, err = updateTonDepsWithDeployedAddresses(deps, ccipSeqReport.Output.Addresses)
+		if err != nil {
+			return sequences.OnChainOutput{}, fmt.Errorf("failed to update TON deps with deployed addresses: %w", err)
+		}
+		// TODO should we include these updates operations in this DeployCCIPSequence ? Probably move to a custom operation and call in CLD ?
+		var txs [][]byte
```

Comment:
Why do our operations return a type like `[][]byte` for a transaction:

- what's the purpose of this?
- can we tackle this immediately and make these op returns a proper type?

Reply:
I think this was introduced because the tlb.InternalMsg cannot be serialized by CLDF, the similar issue we experienced